pkg/apis/serving/v1alpha2/framework_custom.go

```diff
@@ -49,5 +49,8 @@ func (c *CustomSpec) Validate(config *InferenceServicesConfig) error {
 	if err != nil {
 		return fmt.Errorf("Custom container validation error: %s", err.Error())
 	}
+	if c.GetStorageUri() != "" && c.Container.Name != constants.InferenceServiceContainerName {
```

Comment:
Can you take a look at our other validators and implement it in the same style.

Reply:
In other validators, they doesn't need directly define container (we generate container based on all kinds of field), so for custom, it's an exception so that we need validate the container directly.

Reply:
Two things. I think we should just default the container name to our constant name. However, if the user provides a name, we should respect it. As I understand it, this won't have any issue with knative configmaps since the container name is specified by the user.

This also avoids putting weird restrictions on the user's container name. 

What I meant about the error message is typically we expose the error string as a constant and then validate it in the other tests.

Reply:
If a user specifies arbitrarily container name, into which container we should mount model volume in mutating webhook (for now, we mount volume into the container with "user-container" name)?
In fact, for now, if user custom a container with a name and storage Url env, it cannot work (it is an issue)

Reply:
I see. It would be a bit awkward to read the kfservice to grab the container name while mutating pods.